src/routers/StackRouter.js

```diff
@@ -304,6 +304,38 @@ export default (
           };
         }
       }
+
+      if (action.type === NavigationActions.POP) {
+        let numberOfScreens = action.numberOfScreens || 1;
+        if (state.index <= 0) {
+          return state;
+        }
+
+        if (numberOfScreens >= state.routes.length) {
+          numberOfScreens = state.routes.length - 1;
```

Comment:
shouldn't we throw an error here?

Reply:
I went with this approach in case someone doesn't know how many screens there are in the stack. If I was 8 screens deep in a stack and said `pop(9)` I would rather just go to the root vs. deal with an error.

Worth discussion though.

Reply:
how about showing a warning through `console.warn`?